system/jlib/jthread.hpp

```diff
@@ -362,4 +376,26 @@ interface IWorkQueueThread: extends IInterface
 // internally thread persists for specified time waiting before self destroying
 extern jlib_decl IWorkQueueThread *createWorkQueueThread(unsigned persisttime=1000*60);
 
+//--- Functions that manage the current thread state =- for context, tracing, spans etc.
+
+class LogMsgJobInfo;
+
+const LogMsgJobInfo & checkDefaultJobInfo(const LogMsgJobInfo & _jobInfo);
+extern jlib_decl void setDefaultJobId(LogMsgJobId id);
+
+extern const IContextLogger * queryActiveContextLogger();
```

Comment:
wondering for clarity, if this wouldn't be better named queryThreadedContextLogger()

and (see other comment) re. the 2 varieties of "setDefaultJobId" (one in jlog one in jthread) and naming.